consensus/src/pow/helpers.rs

```diff
@@ -62,21 +64,22 @@ where
 /// `difficulty_limit` - found in the PoWChainConfig, as `limit`
 pub fn calculate_new_target(
     actual_timespan_of_last_interval: u64,
-    target_timespan: u64,
+    target_timespan: NonZeroU64,
     old_target: Compact,
     difficulty_limit: Uint256,
 ) -> Result<Compact, ConsensusPoWError> {
     let actual_timespan = Uint256::from_u64(actual_timespan_of_last_interval);
 
-    let target_timespan = Uint256::from_u64(target_timespan);
+    let target_timespan = Uint256::from_u64(target_timespan.get());
 
     let old_target = Uint256::try_from(old_target)
         .map_err(|_| ConsensusPoWError::PreviousBitsDecodingFailed(old_target))?;
 
     // new target is computed by  multiplying the old target by ratio of the actual timespan / target timespan.
     // see Bitcoin's Protocol rules of Difficulty change: https://en.bitcoin.it/wiki/Protocol_rules
-    let mut new_target = old_target * actual_timespan;
-    new_target = new_target / target_timespan;
+    let mut new_target = (old_target * actual_timespan).unwrap_or(Uint256::MAX);
+    new_target =
+        (new_target / target_timespan).expect("target_timespan cannot be zero as per its type");
```

Comment:
Not for this PR: This expect could be avoided if we had a `NonZero` version of the numeric types and an exceptionless division taking a `NonZeroBlah`.

Reply:
It occurred to me yesterday while I was dancing with PoW target time, but that's a huge undertaking and I don't think it's worth it.